### PR TITLE
Update wallet message signing

### PR DESCRIPTION
**Pull request by SarangNoether**
Updates wallet message signing to support:
- hash function domain separation
- signing with either view or spend keys
- binding signatures to full addresses and signing mode

### CODE REVIEW

File: src/cryptonote_config.h [hunk 1]
```diff
@@ -219,6 +219,7 @@ namespace config
   const unsigned char HASH_KEY_RPC_PAYMENT_NONCE = 0x58;
   const unsigned char HASH_KEY_MEMORY = 'k';
   const unsigned char HASH_KEY_MULTISIG[] = {'M', 'u', 'l', 't' , 'i', 's', 'i', 'g', 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+  const char HASH_KEY_MESSAGE_SIGNING[] = "MessageSignature";
```
Comment (TheCharlatan) [inline src/cryptonote_config.h:222]: Since this key will often be used with external systems, I would propose to change this to `MoneroMessageSignature`.

**Comment by Snipa22** [comment]
@SarangNoether Could you rebase this at your convenience please?  Thanks!

File: src/cryptonote_config.h (hunk 1 above)
Comment (SarangNoether) [reply src/cryptonote_config.h:222]: Good idea. Will fix.

**Review by moneromooo-monero — APPROVED** [review]

**Review by selsta — APPROVED** [review]

**Comment by moneromooo-monero** [comment]
That last change weirdly embeds some of my changes, Might go away once rebased though. Gotta be careful they're not undone before merge.

**Pull request merged**